src/controllers/admin/office.ts

```diff
@@ -76,8 +79,94 @@ export const GetAllAdminHandler = async (req: Request, res: Response) => {
       admins: currentAdmins
     });
   } catch (e) {
+    console.log(e)
     return res.status(500).json({
       message: "Failed to retrieve admin details"
     });
   }
 };
+
+export const EditAdminHandler = async (req: Request & { user: { email: string; role: 'Student' | 'Faculty' } }, res: Response) => {
+  try {
+    const validBody = EditAdminValidator.safeParse(req.body);
+    
+    if (!validBody.success) {
+      res.status(400).json({
+        message: "Invalid input data",
+      });
+      throw new CustomError("Invalid input data");
```

Comment:
There is no need for a CustomError() here at the moment. Swap it for an empty return statement.